test/integration-test/src/tests/load.rs

```diff
@@ -154,22 +154,39 @@ fn unload_xdp() {
 fn test_loaded_at() {
     let mut bpf = Bpf::load(crate::TEST).unwrap();
     let prog: &mut Xdp = bpf.program_mut("pass").unwrap().try_into().unwrap();
-    let t1 = SystemTime::now();
-    prog.load().unwrap();
-    let t2 = SystemTime::now();
-    assert_loaded("pass");
-
-    let loaded_at = prog.info().unwrap().loaded_at();
 
-    let range = t1..t2;
+    // The SystemTime::now() calls are not monotonic, which can cause this test to flake. We don't
+    // expect the clock timestamp to continuously jump around, so we add some retries. If the test
+    // is ever correct, we know that the value returned by loaded_at() was reasonable relative to
+    // SystemTime::now().
+    let mut failures = Vec::new();
+    for _ in 0..5 {
+        let t1 = SystemTime::now();
+        prog.load().unwrap();
+        let t2 = SystemTime::now();
+        assert_loaded("pass");
+        let loaded_at = prog.info().unwrap().loaded_at();
+        let range = t1..t2;
+        prog.unload().unwrap();
```

Comment:
shouldn't the assert unloaded be here?

since you're moving things around maybe

let t1 = ...
load()
let t2 = ...
let loaded_at = 

maybe these assert loaded/unloaded calls can all just go away.

Reply:
Done.